### ISSUE TITLE
Help with feature engineering

### ISSUE THREAD

**Issue by swapnens** [swapnens/MachineLearningProject#2]
Hello,

Checking for correlations among features doesn’t require any ML algorithm. We have introduced a few tools for doing it that you can / should use. Of course, you have to reason about the results that you obtain, in order to implement some selection, if applicable.

You can also transform features using PCA, that operates a dimensionality reduction. I briefly mentioned PCA for clustering and used once in an example. With PCA you can transform and rank your features and keep the ones that most explain the variance in your data, meaning the ones that re really relevant for your data:
https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html

You can also check my PCA notebook (I planned to do it in class but didn’t have time….)
https://www.dropbox.com/s/00xxj7t84h013hy/488-S20-PCA.ipynb?dl=0

Regarding polynomial features: well, you should know how to generate them using sklearn methods.

You can also try to generate more general features if you need it
https://scikit-learn.org/stable/modules/preprocessing.html#custom-transformers

In addition, you should be creative. E.g., for the case of images we have extracted so many new different features! From histogram, from gray levels, fro statistical moments, from gradients, from edges, from oriented gradients ….

You should reason on the structure of YOUR representation of an alien DNA, which should not be random, and try to apply transformations that make sense (maybe using statistics). However, you really don’t need to use any additional ML algorithm. 

Best,
Gianni
